src/test-utils/random.test.ts

```diff
@@ -1,34 +1,21 @@
-import { getRandomString, randomiseCase } from './random'
-
-describe('randomiseCase', () => {
-  const longHumanString =
-    "A string long enough that there's a statistically insignificant chance that none of the letters are changed"
-
-  it('randomises randomly', () => {
-    expect(randomiseCase(randomiseCase(longHumanString))).not.toEqual(
-      randomiseCase(longHumanString),
-    )
-  })
-
-  it('makes some letters uppercase', () => {
-    expect(randomiseCase(longHumanString.toLowerCase())).toMatch(/[A-Z]/)
-  })
-})
+import { getRandomString } from './random'
 
 describe('getRandomString', () => {
   it('randomises randomly', () => {
-    expect(getRandomString()).not.toEqual(getRandomString())
-  })
-
-  it('contains uppercase letters', () => {
-    expect(getRandomString()).toMatch(/[A-Z]/)
-  })
-
-  it('contains lowercase letters', () => {
-    expect(getRandomString()).toMatch(/[a-z]/)
-  })
+    // this takes ~580ms
+    for (let i = 0; i < 1000; i++) {
+      const sample = Array(10)
+        .fill(null)
+        .map(() => getRandomString())
+      const isNotSelfSimilar = sample.every(
+        (string, i) => string !== sample[i - 1],
+      )
+      expect(isNotSelfSimilar).toBe(true)
```

Comment:
I ended up smushing all these test cases down into one test that's run over and over because I found the old test would fail unreliably in the fraction of cases where the old algo created a string with (for example) no capital letters. TDDing against this approach seemed much more reliable